service/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/mailgun/service/repo"
+	"net/http"
+	"time"
+)
+
+var (
+	port   string
+	dbConn *repo.DB
+)
+
+func main() {
+	parseArguements()
+
+	dbConn, dbError := repo.GetDB()
+	defer dbConn.Session.Close(context.Background())
+	if dbError != nil {
+		fmt.Println("msg", "Service failed to obtain database connection", "err", dbError)
+		panic("Service failed to obtain database connection")
+	}
+	fmt.Println("msg", "Get database connection complete.")
+
+	h := NewBaseHandler(dbConn.Session)
+
+	router := mux.NewRouter().StrictSlash(true)
+	router.HandleFunc("/events/{domainName}/delivered", h.DeliveredHandler()).Methods("PUT")
+	router.HandleFunc("/events/{domainName}/bounced", h.BouncedHandler()).Methods("PUT")
+	router.HandleFunc("/domains/{domainName}", h.GetDomainHandler()).Methods("GET")
+
+	server := &http.Server{
+		Addr:        ":" + port,
+		Handler:     router,
+		ReadTimeout: 500 * time.Millisecond,
+		IdleTimeout: 90 * time.Second,
+	}
+
+	// here we could determine to use TLS and certs if needed to look something like this
+	//if enableTLS {
+	//	fmt.Println("msg", "Starting Service on TLS", "port", port)
+	//	if err := server.ListenAndServeTLS(certs.CertBasePath+certs.CertPath, certs.CertBasePath+certs.CertKeyPath); err != nil {
+	//		fmt.Println("msg", "Cannot start Service ListenAndServe failed", "err", err.Error())
+	//		panic("Cannot start Service ListenAndServe failed")
+	//	}
+	//}
+
+	if err := server.ListenAndServe(); err != nil {
```

Comment:
If the service needed to be shutdown to redeploy how should it be handled?

Reply:
How we currently handle deployments is we take one of (2 or 3) DCs out of rotation in the region and deploy to that pod. The deployment will update any Kube templates that are needed and restart the pod. 
If we need more control we could add an additional route with admin privileges that could shut down this listen and restart it. This is is convenient but definitely has some security issues. IF we absolutely needed to go that way I would make this a dual operation for two people with some alerting behind it to alert if they're attempts on this route